docs/internal-architecture.md

```diff
@@ -0,0 +1,148 @@
+---
+description: Internal architecture of the project.
+---
+
+# Internal architecture
+
+## Trackers
+
+A tracker is  a process that gets data from an HTTP API or a Blockchain smart contract.
+
+Trackers take placed in the `pkg/tracker` package of the Telliot repository. index tracker is a special type of tracker that creates trackers from an `indexes.json` file.
```

Comment:
I don;t think we even need to mention this. We can mention that there are different type of trackers but no need to specify any actual code implementation details.

Reply:
I've just removed it!